cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/ObjectMappedTableProperties.java

```diff
@@ -40,22 +41,53 @@
   /**
    * The type of object in the table.
    */
-  public static final String TYPE = "type";
+  public static final String OBJECT_TYPE = "object.type";
 
   /**
    * The schema of objects in the table.
    */
-  public static final String SCHEMA = "schema";
+  public static final String OBJECT_SCHEMA = "object.schema";
+
+  /**
+   * The name of the key to use when exploring the table. Defaults to "key".
+   */
+  public static final String EXPLORE_KEY_NAME = "explore.key.name";
+
+  /**
+   * The type of the key when exploring the table. Defaults to bytes.
+   */
+  public static final String EXPLORE_KEY_TYPE = "explore.key.type";
```

Comment:
It is not quite clear what key.name and key.type are used for... In explore, we don't have a key, do we? We only have records. 

Reading the impl of the dataset, I can see that this is the field name that row key in the Table gets mapped to when constructing a record. But the Javadoc for ObjectMappedTable - I believe - does not mention that. Perhaps you can add that?

Also I am wondering what happens if that key is already present as a column in the table? Then we would map both the row key and that column to this field. Which one prevails?